dateparser/conf.py

```diff
@@ -79,8 +81,133 @@ def wrapper(*args, **kwargs):
+    Check if provided settings are valid, if not it raises `SettingValidationError`.
+    Only checks for the modified settings.
+    """
+    settings_values = {
+        'DATE_ORDER': {
+            'values': tuple(date_order_chart.keys()),
+            'type': str,
+        },
+        'TIMEZONE': {
+            'values': (),  # we don't check invalid Timezones as they raise an error
+            'type': str,
+        },
+        'TO_TIMEZONE': {
+            # It defaults to None, but it's not allowed to use it directly
+            'values': (),
+            'type': str
+        },
+        'RETURN_AS_TIMEZONE_AWARE': {
+            # It defaults to 'default', but it's not allowed to use it directly
+            'values': (True, False),
+            'type': bool
+        },
+        'PREFER_DAY_OF_MONTH': {
+            'values': ('current', 'first', 'last'),
+            'type': str
+        },
+        'PREFER_DATES_FROM': {
+            'values': ('current_period', 'past', 'future'),
+            'type': str,
+        },
+        'RELATIVE_BASE': {
+            'values': (),  # unlimited options
+            'type': datetime
+        },
+        'STRICT_PARSING': {
+            'values': (True, False),
+            'type': bool
+        },
+        'REQUIRE_PARTS': {
+            'values': (),  # covered by the 'extra_check'
+            'type': list,
+            'extra_check': _check_require_part
+        },
+        'SKIP_TOKENS': {
+            'values': (),
+            'type': list,
+        },
+        'NORMALIZE': {
+            'values': (True, False),
+            'type': bool
+        },
+        'RETURN_TIME_AS_PERIOD': {
+            'values': (True, False),
+            'type': bool
+        },
+        'PARSERS': {
+            'values': (),  # covered by the 'extra_check'
+            'type': list,
+            'extra_check': _check_parsers
+        },
+        'FUZZY': {
+            'values': (True, False),
+            'type': bool
+        },
+        'PREFER_LOCALE_DATE_ORDER': {
+            'values': (True, False),
+            'type': bool
+        },
+    }
+
+    modified_settings = settings._mod_settings  # check only modified settings
+
+    # check settings keys:
+    for setting in modified_settings:
+        if setting not in settings_values:
+            raise SettingValidationError('"{}" is not a valid setting'.format(setting))
+
+    for setting_name, setting_value in modified_settings.items():
+        setting_type = type(setting_value)
+        setting_props = settings_values[setting_name]
+
+        # check type:
+        if not setting_type == setting_props['type']:
+            raise SettingValidationError(
+                '"{}" must be "{}", not "{}".'.format(
+                    setting_name, setting_props['type'].__name__, setting_type.__name__
+                )
+            )
+
+        # check values:
+        if setting_props['values'] and setting_value not in setting_props['values']:
```

Comment:
What about `setting_props.get('values', None)`, to allow to remove `values` keys above for settings that do not need value validation?